### ISSUE TITLE
Suppressing fields in the generated HTML documenation on a per unit test base.

### ISSUE THREAD

**Issue by wolverine2710** [ScaCap/spring-auto-restdocs#187]
The BIG advantage of Spring Auto Rest Docs (SARD) is the abcense of boilerplating in our unit tests. For Spring Rest Docs (SRD) I have to write between 50-90 lines of boilerplating just to get SRD satisfied wrt to fields in an incoming JSON request and generated JSON response for say a CountryRestControllerTest class containing unit tests. With SARD ZERO lines.

I'm looking for a way to selectively NOT show (suppress variables in the generated HTML documentation output but haven't been able to do so. Is this possible at all?

**Use cases:**
1) Input and output contains an id field - Hibernate domain id. When creating a country in a POST the id is not allowed (because its generated by Hibernate) and we check that. In the output the id has to be shown (JSON and documentation) because its needed later on. 
2) Meta data. Aside from an object (ie Country) We generate metadata in the JSON output. Contains stuff like timestamp,code,status, pagination info which is needed for our frontend. With SRD we show this info once on an overview page in the documentation but do NOT show this for every test. Using The meta data IS ofc shown in the JSON response. 
3) 'Loose' helper get methods NOT connected to member variables. Say a getMonhtlyAccount() method which should NOT be shown in the generated documentation. 

Point (3) I've been able to suppress the output using @JsonIgnore on the getMonhtlyAccount. 

I guess - not tested - that I can suppress the meta data also using a combination of @JsonProperty and @JsonIgnore ([article](http://www.davismol.net/2015/03/21/jackson-using-jsonignore-and-jsonproperty-annotations-to-exclude-a-property-only-from-json-deserialization/)) - IF SARD can hook into this. BUT this is an all or nothing approach and I can't control the output for a specific test. 

My request seems similar to "Support for JsonProperty.Access" [#142](https://github.com/ScaCap/spring-auto-restdocs/issues/142)  but perhaps I've misinterpreted that issue.

I'm hoping there is or will be a way to control the generated HTML output in some way using anontations or perhaps by piggy backing on SRD. With SRD's  FieldDescriptor.ignore() one can control the generated output.

**Comment by jmisur** [comment]
1. This is indeed the same as #142, which is not solved yet
2. Try using `@RestdocsNotExpanded` annotation to suppress output of any field in a class just for documentation purpose.
3. If some field is shown in generated documentation, it's only the field which is actually returned as JSON. We are using the same objectMapper as you use for your controllers (if you configured it properly). E.g. if your objectMapper works on getters, resulting JSON will have all "getter fields" in it. If you want to suppress any getter from result, use @JsonIgnore on it. I expect that if your `getMonhtlyAccount` is shown in a documentation, you have it in response JSON as well.

**Comment by wolverine2710** [comment]
Thanks very much for the reply, unfortunately I´m on holdiday this week and return to work after X-mas. I will try to interpret/digest what you wrote and gonna look if I can work work with that.

**Comment by bessels** [comment]
We had a dev meeting, discussed the POC and the decision was made to switch from SRD to SARD.  SARD is still relative young but the progress made in the last year and the PR's submitted give us a good feeling. Undoubtly we will experience some 'challenges' .....

Wrt "Point (3) I've been able to suppress the output using @jsonignore on the getMonhtlyAccount.". I did not make myself clear it seems. Suppressing this specific getter DOES work.

Should SARD not be able to hook into the relevant parts of Jackson to make #142 work, would it be an option to introduce SARD annontations to suppress fields and anontations for other issues for which there is no workaround.

Wrt hooking into Jackson. Are you perhaps in contact with the Jackson team to see if certain things could be implemented for the SARD use case? That is if its fits in their vison.
